### DISCUSSION TITLE
How to save my own KAMP settings over updates?

### DISCUSSION THREAD

**Discussion by steve8x8** [OpenNeptune3D/OpenNept4une#25] [Q&A] [answered]
When I updated the software (in fluidd's Settings > Software updates), I received a complaint that KAMP_Settings.cfg had been changed - indeed I had tried to address a rather huge blob of filament being dropped from too high rather close to the printed objects, with a non-zero chance to pollute the print area with some leftovers...
I moved the changed file to another name and restored the original one (also checked with "git diff" that evenything would be clean).
I ran the update - only to find that the entire directory content had been rewritten (iow, my file destroyed).
Can this be avoided somehow? AFAIK KAMP is not under the same version control as printer.cfg is. Perhaps it should be...

**Comment by halfmanbear** [comment, marked as answer]
I believe (currently) the OpenNeptune Script reinstalls latest KAMP moonraker etc cfgs when printer.cfg update options is run.

I'll create a dialogue box to prompt the user if they wish to skip this.

If it's kiauh that's borking the confs during updates then I'm not sure on the process fix.

Kamp might just be a symbolic link (default KAMP install) not an actual file in printer-data/config/

**Reply by steve8x8** [reply]
For the time being, I'll keep my diff at a place not affected by the rewrite

**Comment by steve8x8** [comment]
For the time being, I'll keep my diff outside the tree that gets rewritten/replaced ;)
